### PR TITLE
PLT-28 creating workflow for testing

### PR DESCRIPTION
**Pull request by oluwolenpbc**
**JIRA Tickets:**

[PLT-28](https://jira.cms.gov/browse/PLT-28) - Shortcut pipeline for running tests

### What Does This PR Do?

This PR is going to create a workflow with github actions that triggers when users push to a branch on the repo.

### What Should Reviewers Watch For?

Check output in actions

### Usage/Deployment Instructions

Live on the repo once merged to main

### Impacted External Components

n/a

### Database Changes
n/a
### Limitations
n/a
### Security Implications

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [ ] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [ ] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [ ] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [ ] Code checked for PHI/PII exposure

### CODE REVIEW

**Review by devopsmatt — APPROVED** [review]
LGTM

**Review by devopsmatt — APPROVED** [review]
LGTM

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+name: Build and Test
+
+
+on:
+  push:
+    branches:
+      - 'test/*'
+  pull_request:
+    branches:
+      - 'test/*'
```
Comment (gsf) [inline .github/workflows/test.yml:10]: Let's drop the pull_request trigger, only run on push. Also, any reason to limit this to certain branches at this point?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'adopt'
+
+      - name: Set Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          gradle-version: 7.4.2
```
[the remark this answers is not in the document]
Comment (gsf) [reply .github/workflows/test.yml:32]: Also the next step is missing a name

File: build.gradle [hunk 3]
```diff
@@ -4,7 +4,7 @@ import org.gradle.api.tasks.testing.logging.TestLogEvent
 
 plugins {
     id 'java'
-    id 'org.springframework.boot' version "2.7.6"
+    id 'org.springframework.boot' version "3.0.6"
```
Comment (gsf) [inline build.gradle:7]: Is this upgrade required for the test to pass? Looks like this change was made when you hit that authentication issue but this didn't solve that

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+          status=$(curl -s -u ${{ secrets.SONAR_TOKEN }}: "https://sonarqube.cloud.cms.gov/api/qualitygates/project_status?projectKey=ab2d-events" | jq -r .projectStatus.status)
+          echo $status
+          if [ "$status" != "OK" ]; then
+            echo "Quality Gate failed"
+            exit 1
+          fi
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+
+      - name: SBOM
```
Comment (gsf) [inline .github/workflows/test.yml:75]: Looking at the logs on the action, it's unclear if an SBOM is actually being created here. And do we need an SBOM for every test? I would expect one for deploys but one for every test run seems excessive.

File: .github/workflows/test.yml (hunk 1 above)
Comment (gsf) [reply .github/workflows/test.yml:10]: I think my preference would be to run on push to all branches except main https://stackoverflow.com/a/62918562/589391

**Review by gsf — APPROVED** [review]
lgtm

**Review by devopsmatt — APPROVED** [review]
LGTM

**Pull request merged**